### ISSUE TITLE
How to tell FCM to load the firebase-messaging-sw file in a different call route(href)? in @angular/fire (FCM)

### ISSUE THREAD

**Issue by devsleonardo** [angular/angularfire#3506]
How to tell FCM to load the firebase-messaging-sw file in a different call route(href)? in @angular/fire

`"firebase": "^9.1.3" | "@angular/fire": "^17.0.1",`

Let's say in my angular.json I have this initialization href

```
 "baseHref": "/TestRoute/"
```

When I try to make a standard request call, I have tried both methods

```
  requestToken(): void {
    this.afMessaging.requestToken.subscribe(token => {
      console.log(token);
    });
  }
```

or

```
const tempMessaging = await messaging();
  if (tempMessaging) {
    const newSw = await navigator.serviceWorker.register(
      'sw.js'
    );
    return getNotificationPermission(
      async () => await getToken(tempMessaging, { vapidKey, serviceWorkerRegistration:newSw })
    );
  }
```

> ERROR FirebaseError: Messaging: We are unable to register the default service worker. Failed to register a ServiceWorker for scope ('http://localhost:4200/firebase-cloud-messaging-push-scope') with script ('http://localhost:4200/firebase-messaging-sw.js'): A bad HTTP response code (404) was received when fetching the script.

Actually when I try to access the http://localhost:4200/firebase-messaging-sw.js route line it doesn't show anything only if I go to http://localhost:4200/testRoute/firebase-messaging-sw.js

I already tried implementing the `navigator.serviceWorker.register` logic and it didn't work

I think it's important to highlight that I don't want to change the destination folder, but rather the initializing href for calling it.
When I remove the `baseHref` system it works perfectly.

Basically I want...
- from: `http://localhost:4200/firebase-messaging-sw.js`
- to: `http://localhost:4200/testRoute/firebase-messaging-sw.js`

[![enter image description here](https://i.stack.imgur.com/hgpct.png)](https://i.stack.imgur.com/hgpct.png)

[![enter image description here](https://i.stack.imgur.com/puZxN.png)](https://i.stack.imgur.com/puZxN.png)
